### PR TITLE
WFLY-13832 Bug in detecting the org.apache.cxf module

### PR DESCRIPTION
**Pull request by TomasHofman**
Look for cxf module also in exported parent dependencies

https://issues.redhat.com/browse/WFLY-13832

### CODE REVIEW

**Comment by bstansberry** [comment]
@jimma Please review.

File: webservices/server-integration/pom.xml [hunk 1]
```diff
@@ -42,102 +42,107 @@
...
-    <dependency>
-       <groupId>org.wildfly.core</groupId>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
```
Comment (TomasHofman) [inline webservices/server-integration/pom.xml:52]: Only mockito-core was added, rest of the lines were just reformatted...

Comment (jimma) [reply webservices/server-integration/pom.xml:52]: @TomasHofman  Can you please only add the  mockito-core dependency in this PR and format this pom file in another PR ?

File: webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSClassVerificationProcessor.java [hunk 2]
```diff
@@ -138,14 +138,28 @@ private void verifyApacheCXFModuleDependencyRequirement(DeploymentUnit unit) {
...
+    private static boolean hasParentCxfModuleDependency(DeploymentUnit unit) {
+        if (unit.getParent() != null) {
+            final ModuleSpecification moduleSpec = unit.getParent().getAttachment(Attachments.MODULE_SPECIFICATION);
+            for (ModuleDependency dep : moduleSpec.getUserDependencies()) {
+                final String id = dep.getIdentifier().getName();
+                if (cxfExportingModules.contains(id) && dep.isExport()) {
+                    return true;
+                }
+            }
+            return hasParentCxfModuleDependency(unit.getParent());
```
Comment (jimma) [inline webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSClassVerificationProcessor.java:161]: If I understand this correctly, here we already checked Ear deployment. Do we really need this recursion after this check ?  

I don't know if this case has already been checked:  there is another sub-deployment has cxf dependencies, and ear-subdeployments-isolated set to false . Do we need to look at this ?

**Comment by TomasHofman** [comment]
Sorry for delay - working on it.

File: webservices/server-integration/pom.xml (hunk 1 above)
Comment (TomasHofman) [reply webservices/server-integration/pom.xml:52]: OK, done.

File: webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSClassVerificationProcessor.java (hunk 2 above)
Comment (TomasHofman) [reply webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/WSClassVerificationProcessor.java]: Agree, based on jboss deployment structure XSD, recursion is not necessary.

I added another check that should take `subdeployments-isolated` into account.

**Comment by TomasHofman** [comment]
Ready for review.

**Comment by jimma** [comment]
@TomasHofman  Thanks. This looks good to me. @bstansberry

File: webservices/server-integration/src/test/java/org/jboss/as/webservices/deployers/WSClassVerificationProcessorTestCase.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
+package org.jboss.as.webservices.deployers;
```
Comment (darranl) [inline webservices/server-integration/src/test/java/org/jboss/as/webservices/deployers/WSClassVerificationProcessorTestCase.java:1]: Missing copyright header.

**Review by darranl — CHANGES_REQUESTED** [review]
Missing copyright header.

File: webservices/server-integration/src/test/java/org/jboss/as/webservices/deployers/WSClassVerificationProcessorTestCase.java (hunk 3 above)
Comment (TomasHofman) [reply webservices/server-integration/src/test/java/org/jboss/as/webservices/deployers/WSClassVerificationProcessorTestCase.java:1]: Fixed by https://github.com/wildfly/wildfly/pull/13580, thanks!

**Pull request merged**